lib/util/json-writer/editor-config.ts

```diff
@@ -0,0 +1,100 @@
+import { KnownProps, parseString } from 'editorconfig';
+import type { ParseStringResult } from 'editorconfig/src/lib/ini';
+import globToRegExp from 'glob-to-regexp';
+import { logger } from '../../logger/index';
+import { localPathExists, readLocalFile } from '../fs';
+import type { CodeFormat } from './code-format';
+import { IndentationType } from './indentation-type';
+
+export class EditorConfig {
+  private readonly formats: Map<RegExp, KnownProps> = new Map();
+
+  private static instance?: EditorConfig;
+
+  private constructor(parseResult: ParseStringResult = []) {
+    for (const [sectionName, sectionBody] of parseResult) {
+      const regExp = globToRegExp(sectionName ?? '*');
+      this.formats.set(regExp, sectionBody);
+    }
+  }
+
+  public static async getInstance(): Promise<EditorConfig> {
+    if (EditorConfig.instance === undefined) {
+      const editorConfigContent = await this.getEditorConfigContent();
+
+      if (editorConfigContent === undefined) {
+        EditorConfig.instance = new EditorConfig();
+      } else {
+        const parseResult = parseString(editorConfigContent);
+        logger.debug('.editorconfig file successfully parsed');
+        EditorConfig.instance = new EditorConfig(parseResult);
+      }
+    }
+
+    return EditorConfig.instance;
+  }
+
+  public static reset(): void {
+    delete this.instance;
+  }
+
+  public getCodeFormat(fileName: string): CodeFormat {
```

Comment:
I think we should use `EditorConfig.parseFromFiles` api for reading config options

https://www.npmjs.com/package/editorconfig#parsefromfilesfilepath-configs-options

Reply:
Nice idea, eventually I am using `parseSync`